habitat-lab/habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -443,23 +448,42 @@ def _load_navmesh(self, ep_info):
         base_dir = osp.join(*ep_info.scene_id.split("/")[:2])
 
         navmesh_path = osp.join(base_dir, "navmeshes", scene_name + ".navmesh")
-        self.pathfinder.load_nav_mesh(navmesh_path)
 
-        self._navmesh_vertices = np.stack(
-            self.pathfinder.build_navmesh_vertices(), axis=0
-        )
-        self._island_sizes = [
-            self.pathfinder.island_radius(p) for p in self._navmesh_vertices
-        ]
-        self._max_island_size = max(self._island_sizes)
-        self._largest_island_idx = self.pathfinder.get_island(
-            self._navmesh_vertices[np.argmax(self._island_sizes)]
+        if osp.exists(navmesh_path):
+            self.pathfinder.load_nav_mesh(navmesh_path)
```

Comment:
Is it guaranteed that the navmesh that is loaded here has the correct `agent_config.max_climb` and `agent_config.max_slope`? It seems `base_dir` is a scene id, so it could be shared between runs (of different configs) no ?

Reply:
True, it is not guaranteed that the loaded navmesh parameters are correct. However, the user should be able to provide any navmesh they want without risking automated re-compute if a parameter is different. 
The automated re-compute is a backup in case the provided path does not exist. The warning should let the user know this is happening.